worker/src/ocall_bridge/ipfs_ocall.rs

```diff
@@ -15,19 +16,48 @@
 
 */
 
+use crate::ocall_bridge::bridge_api::{Cid, IpfsBridge, OCallBridgeError, OCallBridgeResult};
+use futures::TryStreamExt;
+use ipfs_api::IpfsClient;
+use log::*;
 use std::fs::File;
 use std::io::{Cursor, Write};
-use std::slice;
 use std::str;
 use std::sync::mpsc::channel;
 
-use sgx_types::*;
+pub struct IpfsOCall;
 
-use futures::TryStreamExt;
-use ipfs_api::IpfsClient;
-use log::*;
+impl IpfsBridge for IpfsOCall {
+    fn write_to_ipfs(&self, data: &'static [u8]) -> OCallBridgeResult<Cid> {
+        debug!("    Entering ocall_write_ipfs");
+        Ok(write_to_ipfs(data))
+    }
+
+    fn read_from_ipfs(&self, cid: Cid) -> OCallBridgeResult<()> {
+        debug!("Entering ocall_read_ipfs");
+
+        let result = read_from_ipfs(cid);
+        match result {
+            Ok(res) => {
+                let filename = str::from_utf8(&cid).unwrap();
+                create_file(filename, res).map_err(OCallBridgeError::IpfsError)
+            }
+            Err(_) => Err(OCallBridgeError::IpfsError(
+                "failed to read from IPFS".to_string(),
+            )),
+        }
+    }
+}
 
-pub type Cid = [u8; 46];
+fn create_file(filename: &str, result: Vec<u8>) -> Result<(), String> {
+    match File::create(filename) {
```

Comment:
As we only use a borrow of the result withing the function, we could change the signature to `create_file(..., result: &[u8]) -> ...`

Reply:
Thanks for pointing that out - I should make sure to better recognize where this pattern applies 👍 
(see https://rust-unofficial.github.io/patterns/idioms/coercion-arguments.html )

Reply:
Haha, trust me you're doing quite good already. :)